BUG: (ordqz) always increase parameter lwork by 1.

albop opened the pull request:
I have a testcase where the qz decomposition with reordering (`ordqz`) fails as follows: 
```
  File "/home/pablo/.local/opt/anaconda3/lib/python3.5/site-packages/scipy/linalg/_decomp_qz.py", line 330, in ordqz
    result = tgsen(select, AA, BB, Q, Z, lwork=lwork, liwork=liwork)
ValueError: On entry to DTGSYL parameter number 20 had an illegal value
```
After some investigation, I found out that the parameter lwork given to tgsen, was too small, and that increasing it by one was the minimum to make it work. This is already done when AA and BB are complex.
This PR proposes to do the same when AA and BB are real.

The zip file contains a minimal example to reproduce the bug:
[ordqz_bug.zip](https://github.com/scipy/scipy/files/209137/ordqz_bug.zip)

pv commented:
You can add a test that calls it on 202x202 matrices (the contents of the matrices don't matter, can be randomly generated so no need to ship data files).

albop commented:
@pv: I just added a test with a random matrix. I was not sure it would work for any matrix, since lwork depends on its actual content but after tried with many random matrices, it *seems* to always fail/succeed when they are big enough.
Three comments about the new tests:
- the code I introduced just runs ordqz for bigger matrices but does not check whether the result is correct since this is done somewhere else and I didn't want to refactor everything.
- when ran with argument `sort=iuc`, method `ordqz` used to segfault. But it used to raise the more meaningful above message when a comparison function was given as an argument. I kept this version and left the segfaulting one in comments.
- there seems to be an inconsistency in the doc for the comparison function. The method always accept a method of three arguments `(alphar, alphai, beta)` in the real case but in practice, it always accepts a method of three arguments `(alpha, beta)`. I will open a separate issue for this.

ev-br commented on scipy/linalg/tests/test_decomp.py:
```diff
@@ -2146,6 +2146,39 @@ def test_diff_input_types(self):
+    def test_decompose(self):
+
+        import numpy.random
+        from scipy.linalg import ordqz
```
These imports are not needed here because both are imported already at the module level.

ev-br commented on scipy/linalg/tests/test_decomp.py:
```diff
@@ -2146,6 +2146,39 @@ def test_diff_input_types(self):
+        # for ddtype in [np.float32, np.float64, np.complex, np.complex64]:
+        #     A = numpy.random.random((N,N)).astype(ddtype)
+        #     B = numpy.random.random((N,N)).astype(ddtype)
+        #     [S,T,alpha,beta,U,V] = ordqz(A,B,sort='ouc')
```
On my machine, float32 segfaults but float64 does not. Maybe the right thing to do is to separate this into a test method of its own and decorate it with `@slow`.

ev-br commented:
I confirm both a LAPACK error for `np.float64` matrices / a segfault with `np.float32` matrices on a 64-bit ubuntu precise box, and that both are fixed by this PR.

The patch LGTM modulo a trivial style nitpick and a (non-binding) suggestion on what to do with a test which is currently commented out.

ev-br commented:
I wonder if this is not the root cause of https://github.com/scipy/scipy/issues/5577
(an idle thought, am not going to go back to checking it)

albop commented:
@ev-br : I did the changes you suggest.

ev-br commented:
Seven commits for 40 LOC is a wee bit too much, so I squashed them and merged it in 02d3ab054e7fc62ececfd7509d3ed5a481d55238. (the non-merge commit is 168e98dddbc05a6a76ecf56b608bc9d703884184)

Thanks @albop
